### PR TITLE
Optimize pngs in parallel

### PR DESCRIPTION
**Pull request by leomayleomay**
Closes
https://github.com/ArdanaLabs/Ardana-dApps/issues/49

other changes

- [ ] Every newly introduced function needs to have a [documentation](https://github.com/purescript/documentation/blob/master/language/Syntax.md#comments)
- [ ] The impure tests via `nix run .#offchain:test` were run and all are passing.
- [ ] Where applicable, grammar, punctuation, and spelling within code and comments should be plausibly correct.
- [ ] The issue(s) that the work is addressing should be linked in the PR description.
- [ ] Any significant changes beyond what's clear from the linked issue(s) are documented in the PR description

### CODE REVIEW

**Comment by leomayleomay** [comment]
this is supposed to be a git hook to override existing PNGs in repo instead of generating optimized PNGs in prod only

File: offchain/dusd-ui/flake-module.nix [hunk 1]
```diff
@@ -7,6 +7,27 @@
...
+          {
+            nativeBuildInputs = with pkgs; [ libjxl parallel ];
+          }
+          ''
+            set -euo pipefail
+            mkdir -p $out
+
+            parallel ${escapeShellArgs [ 
+              "--will-cite"
+              "cjxl --quality=100 --effort=9 {} $out/$(basename {})"
```
Comment (toastal) [inline offchain/dusd-ui/flake-module.nix:26]: needs the `.jxl` line

File: offchain/dusd-ui/flake-module.nix [hunk 2]
```diff
@@ -67,6 +94,7 @@
               cp -r ${ui}/lib/node_modules/dusd-ui/build/* $out/
               cp -r ${self'.packages."offchain:dusd-browser"}/dist/* $out/assets/scripts
               cp -r ${font-awesome-sprites}/*.svg $out/assets/images
+              cp -f ${optimized-images}/*.{jxl,png,webp} $out/assets/images
```
Comment (marijanp) [inline offchain/dusd-ui/flake-module.nix:97]: We are copying files around all the time, having them multiple times in the nix store. Can this be a symlink?

Comment (leomayleomay) [reply offchain/dusd-ui/flake-module.nix]: @marijanp great point, now fixed

Comment (leomayleomay) [reply offchain/dusd-ui/flake-module.nix:97]: @marijanp symlink will not work in prod https://636d493b45b2ff125a83dbca--dusd-ui.netlify.app/, once I switched back to copying files, it works https://636d534a201fc923122f0812--dusd-ui.netlify.app/

Comment (toastal) [reply offchain/dusd-ui/flake-module.nix:97]: @marijanp I don't remember who said it, but I had read on the IRC that `cp` is preferred over symlinks the Nix store can better optimize itself not having to follow the indirection.

File: offchain/dusd-ui/netlify.toml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	for = "*.avif"
+	[headers.values]
+		Content-Type = "image/avif"
+		Content-Disposition = "inline"
+
+[[headers]]
+	for = "*.heif"
+	[headers.values]
+		Content-Type = "image/heif"
+		Content-Disposition = "inline"
```
Comment (leomayleomay) [inline offchain/dusd-ui/netlify.toml:11]: do we need these?

**Review by toastal — APPROVED** [review]
Works for me.

File: offchain/dusd-ui/netlify.toml (hunk 3 above)
Comment (toastal) [reply offchain/dusd-ui/netlify.toml:11]: Future-proofing. AVIF would be great to have if we have raster-style images like the "team" section with photographic portraits. HEIF is another image container that could be useful.

**Pull request merged**